### PR TITLE
More robust string escaping and add a path transformation option

### PR DESCRIPTION
**Pull request by wiltzius**
1. Use js-string-escape instead of regex to escape template strings before including in the output file.
2. Add an option to specify a callback function that will transform paths from source location to served location.

What do you think?

### CODE REVIEW

File: src/index.coffee [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 
+  _default_path_transform: (path) ->
+    # Default path transformation is a no-op
+    path
+
   constructor: (config) ->
     @module = config.plugins?.angular_templates?.module or 'templates'
+    @path_transform = config.plugins?.angular_templates?.path_transform or @_default_path_transform
 
   parseHtml: (str) ->
```
Comment (dougmoscrop) [inline src/index.coffee]: I think you can remove this?

**Comment by dougmoscrop** [comment]
Did you accidentally overwrite the @base config part?

**Comment by wiltzius** [comment]
I tried using the @base config, but found it too inflexible so I replaced it with the transform function. I could certainly add back the base as an additional config parameter if you like.

File: src/index.coffee (hunk 1 above)
Comment (wiltzius) [inline src/index.coffee]: Which part? I think it does need to check whether there's a path_transform set in the config and default to using the built-in no-op. Could move the check down to the place it's used but seems better to keep all the config variables together?

Comment (dougmoscrop) [inline src/index.coffee]: The parseHtml function :)

**Comment by dougmoscrop** [comment]
Oh, I gotcha, yeah that makes sense.

I think just that stray parseHtml function can go away now that you're not using it, then it's all good to me.

**Comment by wiltzius** [comment]
Finally had a chance to update; sorry for the long silence. Removed that extraneous parseHtml function. Let me know if there are any other changes you think it needs, or merge if you like.

**Pull request merged**